akka-http-core/src/main/scala/akka/http/impl/engine/ws/FrameOutHandler.scala

```diff
@@ -44,6 +43,9 @@ private[http] class FrameOutHandler(serverSide: Boolean, _closeTimeout: FiniteDu
       case UserHandlerCompleted ⇒
         become(new WaitingForPeerCloseFrame())
         ctx.push(FrameEvent.closeFrame(Protocol.CloseCodes.Regular))
+      case UserHandlerErredOut(ex) ⇒
+        become(new WaitingForPeerCloseFrame())
+        ctx.push(FrameEvent.closeFrame(Protocol.CloseCodes.UnexpectedCondition, "internal error"))
```

Comment:
What was the verdict? (I’ll start running with this approach now, so if you want to alter my course now is exactly the right time)